PythonVisualizations/Array.py

```diff
@@ -468,9 +512,16 @@ def traverse(self, function=print):
     }
     def traverse(self):
         self.startAnimations()
+        
+        #draw an index pointing to the last item in the list 
+        indexDisplay = self.createIndex(len(self.list)-1, 'nItems', level = -1, color = 'black')
+
         callEnviron = self.createCallEnvironment(
             self.traverseCode.strip(), self.traverseCodeSnippets)
         
+        callEnviron |= set(indexDisplay) 

```

Comment:
There's no need to recreate the `nItems` index in each operation.  Use the one built in the `display()` method.